webapp/shop/advantage/views.py

```diff
@@ -625,3 +625,35 @@ def blender_thanks_view(**kwargs):
         "advantage/subscribe/blender/thank-you.html",
         email=kwargs.get("email"),
     )
+
+
+@shop_decorator(area="advantage", permission="user", response="html")
+def activate_magic_attach(advantage_mapper, **kwargs):
+    client_ip = flask.request.headers.get(
+        "X-Real-IP", flask.request.remote_addr
+    )
+    try:
+        activation_status = advantage_mapper.activate_magic_attach(
+            contract_id=flask.request.form.get("contractID"),
+            user_code=flask.request.form.get("userCode").upper(),
+            client_ip=client_ip,
+        )
+        return flask.render_template(
+            "/pro/attach/confirmation.html", status=activation_status
+        )
+    except Exception as e:
+        return flask.render_template(
+            "/pro/attach/confirmation.html",
+            status=json.loads(e.response.content),
+        )
+
+
+@shop_decorator(area="advantage", permission="user", response="html")
+def magic_attach_view(advantage_mapper, **kwargs):
+    user_subscriptions = advantage_mapper.get_user_subscriptions(email=None)
```

Comment:
This function returns subscriptions from the blender marketplace. You could update the `get_user_subscriptions` function to receive a `marketplaces` argument that defaults to ["canonical-ua", "blender"] and from this line you call it with `user_subscriptions = advantage_mapper.get_user_subscriptions(email=None, marketplaces=["canonical-ua"])`